Framework/Muon/test/ApplyMuonDetectorGroupingTest.h

```diff
@@ -181,267 +73,103 @@ class ApplyMuonDetectorGroupingTest : public CxxTest::TestSuite {
-    TS_ASSERT_EQUALS(detids1.size(), 1);
-
-    Mantid::MantidVec vecX1 = ws->getSpectrum(0).readX();
-    Mantid::MantidVec vecX2 = ws->getSpectrum(1).readX();
-    TS_ASSERT_DELTA(vecX1.at(0), 0.000, 0.001);
-    TS_ASSERT_DELTA(vecX1.at(9), 0.900, 0.001);
-    TS_ASSERT_DELTA(vecX2.at(0), 0.000, 0.001);
-    TS_ASSERT_DELTA(vecX2.at(9), 0.900, 0.001);
-
-    Mantid::MantidVec vecY1 = ws->getSpectrum(0).readY();
-    Mantid::MantidVec vecY2 = ws->getSpectrum(1).readY();
-    TS_ASSERT_DELTA(vecY1.at(0), 0, 0.1);
-    TS_ASSERT_DELTA(vecY1.at(9), 9, 0.1);
-    TS_ASSERT_DELTA(vecY2.at(0), 10, 0.1);
-    TS_ASSERT_DELTA(vecY2.at(9), 19, 0.1);
-
-    Mantid::MantidVec vecE1 = ws->getSpectrum(0).readE();
-    Mantid::MantidVec vecE2 = ws->getSpectrum(1).readE();
-    TS_ASSERT_DELTA(vecE1.at(0), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE1.at(9), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE2.at(0), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE2.at(9), 0.005, 0.0001);
-  }
-
-  void test_createAsymmetryWorkspace() {
-
-    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(5, 10);
-    AnalysisDataService::Instance().addOrReplace("inputData", ws);
+  void test_algorithm_executes_with_default_arguments() {
 
-    TS_ASSERT_EQUALS(ws->getNumberHistograms(), 5);
-
-    const std::set<detid_t> detids0 = ws->getSpectrum(0).getDetectorIDs();
-    const std::set<detid_t> detids1 = ws->getSpectrum(1).getDetectorIDs();
-
-    TS_ASSERT_EQUALS(*detids0.begin(), 1);
-    TS_ASSERT_EQUALS(*detids1.begin(), 2);
-
-    TS_ASSERT_EQUALS(detids0.size(), 1);
-    TS_ASSERT_EQUALS(detids1.size(), 1);
+    auto ws = createCountsWorkspace(5, 10, 0.0);
+    setUpADSWithWorkspace setup(ws);
+    Mantid::API::IAlgorithm_sptr alg =
+        algorithmWithPropertiesSet("inputData", "inputGroup");
 
-    Mantid::MantidVec vecX1 = ws->getSpectrum(0).readX();
-    Mantid::MantidVec vecX2 = ws->getSpectrum(1).readX();
-    TS_ASSERT_DELTA(vecX1.at(0), 0.000, 0.001);
-    TS_ASSERT_DELTA(vecX1.at(9), 0.900, 0.001);
-    TS_ASSERT_DELTA(vecX2.at(0), 0.000, 0.001);
-    TS_ASSERT_DELTA(vecX2.at(9), 0.900, 0.001);
+    TS_ASSERT_THROWS_NOTHING(alg->execute());
+    TS_ASSERT(alg->isExecuted());
+  }
 
-    Mantid::MantidVec vecY1 = ws->getSpectrum(0).readY();
-    Mantid::MantidVec vecY2 = ws->getSpectrum(1).readY();
-    TS_ASSERT_DELTA(vecY1.at(0), 220.0, 0.1);
-    TS_ASSERT_DELTA(vecY1.at(9), 152.8, 0.1);
-    TS_ASSERT_DELTA(vecY2.at(0), 420.0, 0.1);
-    TS_ASSERT_DELTA(vecY2.at(9), 285.6, 0.1);
+  void test_output_produced_in_ADS_for_counts_analysis() {
 
-    Mantid::MantidVec vecE1 = ws->getSpectrum(0).readE();
-    Mantid::MantidVec vecE2 = ws->getSpectrum(1).readE();
-    TS_ASSERT_DELTA(vecE1.at(0), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE1.at(9), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE2.at(0), 0.005, 0.0001);
-    TS_ASSERT_DELTA(vecE2.at(9), 0.005, 0.0001);
+    auto ws = createCountsWorkspace(5, 10, 0.0);
+    setUpADSWithWorkspace setup(ws);
```

Comment:
I don't understand this as I cannot see where setup goes.

Reply:
What do you mean?

Reply:
to me it looks like the object is setUpADSWithWorkspace  and the function is setup

Reply:
The object is `setUpADSWithWorkspace` (defined in namespace at the top), the instance is called `setup` which takes the workspace as an argument and puts it in the ADS, as well as performing a few other common tasks for testing the algorithm.